src/mpags-cipher.cpp

```diff
@@ -87,69 +83,50 @@ int main(int argc, char* argv[])
     if (!inputFile.empty()) {
-        std::cerr << "[warning] input from file ('" << inputFile
-                  << "') not implemented yet, using stdin\n";
-    }
-
-    // loop over each character from user input
-    while (std::cin >> inputChar) {
-        // Uppercase alphabetic characters
-        if (std::isalpha(inputChar)) {
-            inputText += std::toupper(inputChar);
-            continue;
+        std::ifstream in_file {inputFile};
+        bool ok_to_read = in_file.good();
+        if (!ok_to_read) {
+            std::cerr << "[error] failed to open the input file" << std::endl;
+            return 1;
         }
-
-        // Transliterate digits to English words
-        switch (inputChar) {
-            case '0':
-                inputText += "ZERO";
-                break;
-            case '1':
-                inputText += "ONE";
-                break;
-            case '2':
-                inputText += "TWO";
-                break;
-            case '3':
-                inputText += "THREE";
-                break;
-            case '4':
-                inputText += "FOUR";
-                break;
-            case '5':
-                inputText += "FIVE";
-                break;
-            case '6':
-                inputText += "SIX";
-                break;
-            case '7':
-                inputText += "SEVEN";
-                break;
-            case '8':
-                inputText += "EIGHT";
-                break;
-            case '9':
-                inputText += "NINE";
-                break;
+        while (in_file >> inputChar) {
+            inputText += transformChar(inputChar);
+        }
+    } else {
+        // loop over each character from user input
+        while (std::cin >> inputChar) {
+            inputText += transformChar(inputChar);
+            // If the character isn't alphabetic or numeric, DONT add it
         }
-
-        // If the character isn't alphabetic or numeric, DONT add it
     }
 
-    // Print out the transliterated text
-
     // Warn that output file option not yet implemented
     if (!outputFile.empty()) {
-        std::cerr << "[warning] output to file ('" << outputFile
-                  << "') not implemented yet, using stdout\n";
+        std::ofstream out_file {outputFile};
+        bool ok_to_write = out_file.good();
+        if (!ok_to_write) {
+            std::cerr << "[error] failed to write to the output file" << std::endl;
+        }
+        // Run encrption/decryption if needed
+        if (cipherTrigger != 0) {
+            outputText += runCaesarCipher(inputText, key, encrypt);
+            out_file << outputText;
+        } else {
+            out_file << inputText;
+        }
+    } else {
+        if (cipherTrigger != 0) {
+            outputText += runCaesarCipher(inputText, key, encrypt);
+            std::cout << outputText << std::endl;
+        } else {
+            std::cout << inputText << std::endl;
+        }
```

Comment:
Would be better to do the ciphering before this if...else... because those lines are identical in each block.  Always best to avoid repeating code.